crime-applications-adaptor/api-tests/src/test/java/uk/gov/justice/laa/crime/applications/adaptor/apispecification/RequestSpecificationBuilder.java

```diff
@@ -38,18 +38,28 @@ public class RequestSpecificationBuilder {
   private RequestSpecificationBuilder() {
   }
 
-  public static RequestSpecification getCAMReqSpec() {
-    RequestSpecBuilder requestSpecBuilder = setUpRequestSpecBuilder(CAM_BASE_URL);
+  public static RequestSpecification getCAMCrimeApplyReqSpec() {
+    return getCAMReqSpec(CAM_BASE_URL, CAM_JWT_ISSUER, CAM_JWT_SECRET);
+  }
+
+  public static RequestSpecification getCAMReqSpec(String baseUrl, String jwtIssuer, String jwtSecret) {
+    RequestSpecBuilder requestSpecBuilder = setUpRequestSpecBuilder(baseUrl);
     requestSpecBuilder.addHeader("Authorization",
-        "Bearer " + JwtUtil.generateJwt(CAM_JWT_ISSUER, CAM_JWT_SECRET));
+        "Bearer " + JwtUtil.generateJwt(jwtIssuer, jwtSecret));
     return requestSpecBuilder.build();
   }
 
-  public static RequestSpecification getCAAReqSpec() {
-    RequestSpecBuilder requestSpecBuilder = setUpRequestSpecBuilder(CAA_BASE_URL);
+  public static RequestSpecification getCAACrimeApplyReqSpec() {
+    return getCAAReqSpec(CAA_BASE_URL, CAA_OAUTH_BASE_URL, CAA_OAUTH_CLIENT_ID, CAA_OAUTH_CLIENT_SECRET,
+            CAA_OAUTH_TOKEN_URI);
+  }
+
+  public static RequestSpecification getCAAReqSpec(String baseUrl, String authUrl,
```

Comment:
It looks like `getCAAReqSpec` is only used in one place by `getCAACrimeApplyReqSpec` so I don't think it needs to be in it's own method, instead inline it and have a single method `getCAACrimeApplyReqSpec`.

Also maybe rename `getCAACrimeApplyReqSpec` to `getCrimeApplyAdapterReqSpec` without the duplicated CAA and CrimeApplyAdapter. ()I would also ask te rename with `RequestSpecification` or just `Request` but I can see many other method names also follow this pattern and unless you are prepared to change them all, we should stick with `ReqSpec`.

Reply:
Hi Alex. This is a change I made in response to Mike's ticket. You can see the discussion [here](https://github.com/ministryofjustice/laa-crime-applications-adaptor/pull/144#discussion_r1560926603).